### PR TITLE
otherProperties implementation

### PR DESCRIPTION
**Pull request by fabianvf**
@chrisseto @erinspace does this seem sane?
# Ported
- [x] arxiv.py
- [x] asu.py
- [x] bhl.py
- [x] biomed.py
- [x] calhoun.py
- [x] calpoly.py
- [x] caltech.py
- [x] clinicaltrials.py
- [x] cmu.py
- [x] cogprints.py
- [x] columbia.py
- [x] crossref.py
- [x] dataone.py
- [x] doepages.py
- [x] figshare.py
- [x] mit.py
- [x] opensiuc.py
- [x] osf.py
- [x] plos.py
- [x] pubmedcentral.py
- [x] scitech.py
- [x] spdataverse.py
- [x] stcloud.py
- [x] stepic.py
- [x] tdar.py
- [x] texasstate.py
- [x] trinity.py
- [x] ucescholarship.py
- [x] uiucideals.py
- [x] upennsylvania.py
- [x] utaustin.py
- [x] uwashington.py
- [x] valposcholar.py
- [x] vtech.py
- [x] waynestate.py

# Analyzed
- [x] arxiv.py
- [x] asu.py
- [x] bhl.py
- [x] biomed.py
- [x] calhoun.py
- [x] calpoly.py
- [x] caltech.py
- [x] clinicaltrials.py
- [x] cmu.py
- [x] cogprints.py
- [x] columbia.py
- [x] crossref.py
- [x] dataone.py
- [x] doepages.py
- [x] figshare.py
- [x] mit.py
- [x] opensiuc.py
- [x] osf.py
- [x] plos.py
- [x] pubmedcentral.py
- [x] scitech.py
- [x] spdataverse.py
- [x] stcloud.py
- [x] stepic.py
- [x] tdar.py
- [x] texasstate.py
- [x] trinity.py
- [x] ucescholarship.py
- [x] uiucideals.py
- [x] upennsylvania.py
- [x] utaustin.py
- [x] uwashington.py
- [x] valposcholar.py
- [x] vtech.py
- [x] waynestate.py

### CODE REVIEW

File: scrapi/base/helpers.py [hunk 1]
```diff
@@ -1,14 +1,42 @@
...
+    for arg in args:
+        name, expr = arg[0], arg[1]
+        kwargs = arg[2] if len(arg) > 2 else {}
+        description, uri = kwargs.get('description'), kwargs.get('uri')
+        ret.append(build_property(name, expr, description=description, uri=uri))
+    return ret
+
+
+def build_property(name, expr, description=None, uri=None):
+    from scrapi.base.schemas import CONSTANT
```
Comment (chrisseto) [inline scrapi/base/helpers.py]: Why is this imported in the function?

File: scrapi/base/helpers.py [hunk 2]
```diff
@@ -1,14 +1,42 @@
...
+
+    property = {
```
Comment (chrisseto) [inline scrapi/base/helpers.py]: `property` is a builtin, please rename.

File: scrapi/base/helpers.py [hunk 3]
```diff
@@ -98,3 +126,10 @@ def oai_process_contributors(*args):
 
 def pack(*args, **kwargs):
     return args, kwargs
+
+
+def language_code(language):
+    try:
+        return languages.get(name=language)
```
Comment (chrisseto) [inline scrapi/base/helpers.py]: No default?

File: scrapi/base/transformer.py [hunk 4]
```diff
@@ -26,20 +26,29 @@ def transform(self, doc):
...
-                transformed[key] = value(doc)
+            transformed[key] = self._transform_value(value, doc)
         return transformed
 
-    def _transform_iterable(self, l, doc):
-
-        if isinstance(l[0], tuple) and len(l) == 2:
-            return self._transform_args_kwargs(l, doc)
+    def _transform_value(self, value, doc):
+        if isinstance(value, dict):
```
Comment (chrisseto) [inline scrapi/base/transformer.py]:
```python
{
    dict: self._transform_dict,
    tuple: self._transform_tuple,
    ...
}
```

File: scrapi/base/helpers.py (hunk 3 above)
Comment (fabianvf) [inline scrapi/base/helpers.py]: Not so far as I could tell.

File: scrapi/base/transformer.py (hunk 4 above)
Comment (fabianvf) [inline scrapi/base/transformer.py]: Doesn't work for the args/kwargs case.

**Comment by erinspace** [comment]
Woop! Looks good to me. :+1: :+1:

**Pull request merged**